src/patch.js

```diff
@@ -22,6 +22,33 @@ var walker = require('./walker'),
     getWalker = walker.getWalker,
     setWalker = walker.setWalker;
 
+/**
+ * @const {boolean}
+ */
+var IS_PRODUCTION = process.env.NODE_ENV === 'production';
+
+if (!IS_PRODUCTION) {
+  var assertNoUnclosedTags = function(root) {
```

Comment:
It was in the sense of a template corresponding to a single function. For example, you could do:

```
function t1() {
  elementOpen('div');
}

function t2() {
  elementClose('div');
}

function render() {
  t1();
  t2();
}
```

A given template might not be balanced, but a patch operation still must be.